Jenkinsfile

```diff
@@ -1,213 +1,252 @@
-				 if [[ $cmd =~ $two_proc_test_pattern ]]; then
-				 	 launcher=$launcher2
-				 else
-					 launcher=$launchern
-				 fi
-				 (set -x; eval $launcher $cmd)	# Doesn't work without eval!
-			    done
-		       )
-		       (
-		       #ompi - mpi stress test
-		        install_path="/home/build/jenkinsbuild/workspace/libfabrics-pipbuild/"
-		       	$install_path/ompi/bin/mpirun -np 16 --host n105:8,n107:8 --mca mtl_ofi_provider_include psm2 --mca orte_base_help_aggregate 0 --mca mtl ofi --mca pml cm -tag-output $install_path/ompi/stress/mpi_stress -dcr
-		       )	
-			
-			echo "The return status of runfabtests.sh is :"
-                         echo $?
+   stage('parallel-fi_getinfo-stage') {
+            parallel {
+                stage('eth-test') {
+                     agent {
+                        node {
+                            label 'eth'
+                          
+                        }
+                     }
+                     steps{
+                        withEnv(['PATH+EXTRA=/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/bin/:$PYTHONPATH']) {
+                          sh """
+                            env
+                            echo "run fi_info stage"                   
+                            echo ${env.NODE_NAME}
+                            echo ${env.FABRIC}
+                            echo ${env.CI_SITE_CONFIG}
+     
+                            ${env.ofi_install_dir}/${env.BRANCH_NAME}/${env.BUILD_NUMBER}/bin/fi_info -f ${env.FABRIC}
+                                                        
+        python3.7 ${env.WORKSPACE}/contrib/Intel/JenkinsBuildScripts/run.py tcp n1,n2 fabtests
+        python3.7 ${env.WORKSPACE}/contrib/Intel/JenkinsBuildScripts/run.py udp n1,n2 fabtests
+        python3.7 ${env.WORKSPACE}/contrib/Intel/JenkinsBuildScripts/run.py udp n1,n2 fabtests --util=rxd
```

Comment:
I think the provider name should be passed as "udp;ofi_rxd" to be consistent with how it's used in other places. I suppose the node names would have to be got from somewhere else instead of hardcoding here?

Does Jenkinsfile allow some sort of loop to avoid repetition?

Reply:
the python script internally handles this. the whole thing is written as a part of a shell script inside the steps block so it should be possible using the shell script for loops. 
The Node names can be got from an inbuilt jenkins variable called NODE_NAME. which holds the name specified in the name field when creating the nodes. I had kept the names as nXX-fabric e.g. n65-mlx5 so that its more descriptive, but we can either define another env variable or strip the fabric prefix or just find another way to add the description ( I checked this though and it wasn't there without installing additional plugin)

Reply:
I was thinking of moving the loop within the python script but does having the loop / individual steps in Jenkinsfile allows better granularity while using the UI?

We can keep the node names as nXX-fabric, but one would have to take care to strip the fabric before ssh-ing. Either way, the node name would have to be got from the env or the "executor's properties".